### DISCUSSION TITLE
Validation error on custom unique index?

### DISCUSSION THREAD

**Discussion by denisgurev** [pocketbase/pocketbase#2054] [Q&A] [answered]
I use the workaround from https://github.com/pocketbase/pocketbase/issues/544 so that I can have a combined unique check.

It works fine but the err.data from the response is empty and I have to implement different error handling only for this case which is not ideal.

Is it possible to return error response similar to the one when nonempty is checked (with code and message attributes for the invalid fields)?

**Comment by ganigeorgiev** [comment, marked as answer]
At the moment you can't (without overwriting the API response and manually parsing the original `ApiError.RawData()` inside the [`OnBeforeApiError` hook](https://pocketbase.io/docs/event-hooks/#app-hooks)).

I've already implemented it locally as part of the indexes tasks (https://github.com/pocketbase/pocketbase/issues/345, https://github.com/pocketbase/pocketbase/issues/544) and will be available with the next v0.14.0 release (_there are no ETAs yet_). I'll try to push it in the `develop` branch once I finalize the UI and the automated tests.

Note that it will work only with regular column indexes (eg. `create [unique] index foo on bar (col1, col2, ...) [where ...]`).
Indexes on expressions (eg. `create [unique] index foo on bar (strftime(...)) [where ...]`) are not checked because there is no easy and reliable way to resolve the column(s) participating in the expression without implementing a full-blown sql parser. In those cases you still will get a generic non-field related error, but I think those are rare.

**Reply by denisgurev** [reply]
Thank you for the speedy reply. I'll wait for v0.14. My use case is the same as #544 there are no expression, functions or anything complex like that.

**Reply by denisgurev** [reply]
Bit off topic, but just saw a notification that #544 is already implemented, if I have created manually a unique index will I have to recreate it later?

btw. I really like the small UI changes. Can't wait for the release.

**Reply by ganigeorgiev** [reply]
> if I have created manually a unique index will I have to recreate it later?

No, it shouldn't be necessary. There is a system migration that will take care for this - https://github.com/pocketbase/pocketbase/blob/develop/migrations/1679986800_add_indexes_column.go
